DOC-2194: TinyMCE 6.7.2 Release notes. (#2941)
tinymce/tinymce-docs: modules/ROOT/pages/6.7.2-release-notes.adoc

= TinyMCE 6.7.2
:navtitle: TinyMCE 6.7.2
:description: Release notes for TinyMCE 6.7.2
:keywords: releasenotes, new, changes, bugfixes
:page-toclevels: 1

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.7.2 was released for {enterpriseversion} and {cloudname} on Wednesday, November 1^st^, 2023. These release notes provide an overview of the changes for {productname} 6.7.2, including:

> * xref:accompanying-premium-skins-and-icon-packs-changes[Accompanying Premium Skins and Icon Packs changes]
> * xref:bug-fixes[Bug fixes]
> 
> 
> [[accompanying-premium-skins-and-icon-packs-changes]]
> == Accompanying Premium Skins and Icon Packs changes
> 
> The {productname} 6.7.2 release includes an accompanying release of the **Premium Skins and Icon Packs**.
> 
> === Premium Skins and Icon Packs
> 
> The **Premium Skins and Icon Packs** release includes the following updates:
> 
> The **Premium Skins and Icon Packs** were rebuilt to pull in the changes also incorporated into the default {productname} 6.7.2 skin, Oxide.
> 
> For information on using premium skins and icon packs, see: xref:premium-skins-and-icons.adoc[Premium Skins and Icon Packs].

Author: ```suggestion
* xref:bug-fixes[Bug fixes]
```

As a result, the editor no longer renders the flicker previously displayed when the editor is refreshed.

=== Toggling a list that contained a list item element — `<li>` — which, in turn, contained another list item element as its first child, removed other content within the first list item element.
// TINY-10213
When a user attempted to indent or outdent a list element inside a nested list, the list was converted into a model which resulted in it being recreated from that model.

As a consequence of this, the model would not handle a list and a not-list items inside a `<li>` element, which resulted in content deletion of the not-list item inside the `<li>` structures like:

[source, html]
----
<li>
  [some_list]
  [some_other_element]
</li>
----

In addition, when the editor would try to manage comments at the `<li>` element level, a console error would be thrown resulting in all other elements being ignored.

{productname} 6.7.2 addresses this issue, now, our model correctly manages comments and other elements when they are preceded from a list inside an `li`, and the errors are no longer present.

> 
> === If multiple inline formatting options were set at the insertion point, turning any one of these options off turned all the set options off.
> //TINY-10132
> 
> // CCFR

Author: ```suggestion
```

* 
*
*
*
////
Example ONLY
The Java server-side components have been updated to the following versions:

* `ephox-hyperlinking.war`: 2.105.13
* `ephox-image-proxy.war`: 2.106.4
* `ephox-spelling.war`: 2.118.6
////
For information on:

////
Example ONLY 
* The Enhanced Media Embed plugin, see: xref:introduction-to-mediaembed.adoc[Enhanced Media Embed plugin].
*
*
*
////
=== Updating the self-hosted server-side components
////
The new versions of the server-side services provide updates for the Java-based server-side components. To deploy the updated version of the server-side components:

. Update your Java Application Server to the minimum required version:

include::partial$misc/supported-application-servers.adoc[]

. Replace the existing server-side `.war` files with the `.war` files bundled with {productname} 6.2 or later.

For information on:

* Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
* Deploying the server-side components using Docker, see: xref:bundle-intro-setup.adoc[Containerized service deployments].

> ////

Author: @frossi933 Ive added the initial placeholder with some examples, based on the last 6.2 service side changes.

> [[accompanying-premium-self-hosted-server-side-component-changes]]
> == Accompanying Premium self-hosted server-side component changes
> 
> The {productname} 6.7.2 release includes accompanying changes affecting the {productname} **self-hosted** services for the following plugins:
> 
> * 
> *
> *
> *
> ////
> Example ONLY
> The Java server-side components have been updated to the following versions:
> 
> * `ephox-hyperlinking.war`: 2.105.13
> * `ephox-image-proxy.war`: 2.106.4
> * `ephox-spelling.war`: 2.118.6
> ////
> For information on:
> 
> ////
> Example ONLY 
> * The Enhanced Media Embed plugin, see: xref:introduction-to-mediaembed.adoc[Enhanced Media Embed plugin].
> *
> *
> *
> ////
> === Updating the self-hosted server-side components
> ////
> The new versions of the server-side services provide updates for the Java-based server-side components. To deploy the updated version of the server-side components:
> 
> . Update your Java Application Server to the minimum required version:
> 
> include::partial$misc/supported-application-servers.adoc[]
> 
> . Replace the existing server-side `.war` files with the `.war` files bundled with {productname} 6.2 or later.
> 
> For information on:
> 
> * Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
> * Deploying the server-side components using Docker, see: xref:bundle-intro-setup.adoc[Containerized service deployments].
> ////

Author: ```suggestion
[[accompanying-premium-self-hosted-server-side-component-changes]]
== Accompanying Premium self-hosted server-side component changes

The {productname} 6.7.3 release includes accompanying changes affecting the {productname} **self-hosted** services for the following plugins:

* Enhanced Media Embed plugin (`mediaembed`)
* Export plugin (`export`)
** FIXED: https://www.cve.org/CVERecord?id=CVE-2023-44487[CVE-2023-44487] for Netty. Upgrading to version 4.1.100.Final
** Service codebase now it's based on Scala 2.13.12 (instead of Scala 2.12.16).
* Enhanced Image Editing plugin (`editimage`)
* Link Checker plugin (`linkchecker`)
** FIXED: https://www.cve.org/CVERecord?id=CVE-2023-44487[CVE-2023-44487] for Netty. Upgrading to version 4.1.100.Final
** Service codebase now it's based on Scala 2.13.12 (instead of Scala 2.12.16)
** Server side cache implementation was refactored to use Scaffeine, a thin Scala wrapper for Caffeine which is very well known and stable Java caching library offering a great performance and support
** Include code refactors to improve code abstraction and encapsulation
* Spell Checker Pro plugin (`tinymcespellchecker`)
* Spelling Autocorrect plugin (`autocorrect`)
** Service codebase now it's based on Scala 2.13.12 (instead of Scala 2.12.16).
** Major code refactors to improve maintainability and performance.
** Server side cache implementation was refactored to use Scaffeine, a thin Scala wrapper for Caffeine which is very well known and stable Java caching library offering a great performance and support.
** Service log level was changed from DEBUG to INFO, to reduce logs verbosity and gain more clarity.

For information on:

* The Enhanced Media Embed plugin, see: xref:introduction-to-mediaembed.adoc[Enhanced Media Embed plugin].
* The Export plugin, see: xref:export.adoc[Export plugin].
* The Enhanced Image Editing plugin, see: xref:editimage.adoc[Enhanced Image Editing plugin].
* The Link Checker plugin, see: xref:linkchecker.adoc[Link Checker plugin].
* The Spell Checker Pro plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro plugin].
* The Spelling Autocorrect plugin, see: xref:autocorrect.adoc[Spelling Autocorrect plugin].
* Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].

The Java server-side components have been updated to the following versions:

* `ephox-hyperlinking.war`: pending
* `ephox-image-proxy.war`: pending
* `ephox-spelling.war`: pending

=== Updating the self-hosted server-side components

The new versions of the server-side services provide updates for the Java-based server-side components. To deploy the updated version of the server-side components:

. Update your Java Application Server to the minimum required version:

include::partial$misc/supported-application-servers.adoc[]

. Replace the existing server-side `.war` files with the `.war` files bundled with {productname} 6.7.2 or later.

For information on:

* Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
* Deploying the server-side components using Docker, see: xref:bundle-intro-setup.adoc[Containerized service deployments].

include::partial$misc/admon-no-functionality-changes-in-updated-server-side-components.adoc[]
```

include::partial$misc/supported-application-servers.adoc[]

. Replace the existing server-side `.war` files with the `.war` files bundled with {productname} 6.7.2 or later.

For information on:

* Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
* Deploying the server-side components using Docker, see: xref:bundle-intro-setup.adoc[Containerized service deployments].

include::partial$misc/admon-no-functionality-changes-in-updated-server-side-components.adoc[]

[[bug-fixes]]
== Bug fixes

{productname} 6.7.2 also includes the following bug fixes:

=== The function `getModifierState` did not work on events passed through the editor as expected.
// TINY-10263
The exception error "Uncaught TypeError: Illegal invocation" was displayed in the console, when an integrator called the 'getModifierState' function on a **keydown** event object.

As a consequence, any attempts to invoke this function resulted in the TypeError being displayed in the console.

{productname} 6.7.2 addresses this, now, when a integrator uses the `getModifierState` on the event object a keydown event no longer triggers an error.

> 

Author: ```suggestion
```

= TinyMCE 6.7.2
:navtitle: TinyMCE 6.7.2
:description: Release notes for TinyMCE 6.7.2
:keywords: releasenotes, new, changes, bugfixes
:page-toclevels: 1

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.7.2 was released for {enterpriseversion} and {cloudname} on Wednesday, November 1^st^, 2023. These release notes provide an overview of the changes for {productname} 6.7.2, including:

* xref:accompanying-premium-self-hosted-server-side-component-changes[Accompanying Premium self-hosted server-side component changes]
* xref:bug-fixes[Bug fixes]
* xref:security-fixes[Security fixes]

[[accompanying-premium-self-hosted-server-side-component-changes]]
== Accompanying Premium self-hosted server-side component changes

> The {productname} 6.7.3 release includes accompanying changes affecting the {productname} **self-hosted** services for the following plugins:

Author: ```suggestion
The {productname} 6.7.2 release includes accompanying changes affecting the {productname} **self-hosted** services for the following plugins:
```

The new versions of the server-side services provide updates for the Java-based server-side components. To deploy the updated version of the server-side components:

. Update your Java Application Server to the minimum required version:

include::partial$misc/supported-application-servers.adoc[]

. Replace the existing server-side `.war` files with the `.war` files bundled with {productname} 6.7.2 or later.

For information on:

* Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
* Deploying the server-side components using Docker, see: xref:bundle-intro-setup.adoc[Containerized service deployments].

include::partial$misc/admon-no-functionality-changes-in-updated-server-side-components.adoc[]

[[bug-fixes]]
== Bug fixes

{productname} 6.7.2 also includes the following bug fixes:

=== The function `getModifierState` did not work on events passed through the editor as expected.
// TINY-10263
The exception error "Uncaught TypeError: Illegal invocation" was displayed in the console, when an integrator called the 'getModifierState' function on a **keydown** event object.

> 

Author: ```suggestion
```

= TinyMCE 6.7.2
:navtitle: TinyMCE 6.7.2
:description: Release notes for TinyMCE 6.7.2
:keywords: releasenotes, new, changes, bugfixes
:page-toclevels: 1

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.7.2 was released for {enterpriseversion} and {cloudname} on Wednesday, November 1^st^, 2023. These release notes provide an overview of the changes for {productname} 6.7.2, including:

* xref:accompanying-premium-self-hosted-server-side-component-changes[Accompanying Premium self-hosted server-side component changes]
* xref:bug-fixes[Bug fixes]
* xref:security-fixes[Security fixes]

[[accompanying-premium-self-hosted-server-side-component-changes]]
== Accompanying Premium self-hosted server-side component changes

The {productname} 6.7.3 release includes accompanying changes affecting the {productname} **self-hosted** services for the following plugins:

> * For **Enhanced Media Embed** plugin `mediaembed` and the **Export** plugin `export`.
> ** FIXED: https://nvd.nist.gov/vuln/detail/CVE-2023-44487[CVE-2023-44487] for Netty. Upgrading to version 4.1.100.Final
> ** Service codebase now it's based on Scala 2.13.12 (instead of Scala 2.12.16).
> * For **Enhanced Image Editing** plugin `editimage` and the **Link Checker** plugin `linkchecker`.
> ** FIXED: https://nvd.nist.gov/vuln/detail/CVE-2023-44487[CVE-2023-44487] for Netty. Upgrading to version 4.1.100.Final
> ** Service codebase now it's based on Scala 2.13.12 (instead of Scala 2.12.16)
> ** Server side cache implementation was refactored to use Scaffeine, a thin Scala wrapper for Caffeine which is very well known and stable Java caching library offering a great performance and support
> ** Include code refactors to improve code abstraction and encapsulation
> * For **Spell Checker Pro** plugin `tinymcespellchecker` and the **Spelling Autocorrect** plugin `autocorrect`.
> ** Service codebase now it's based on Scala 2.13.12 (instead of Scala 2.12.16).
> ** Major code refactors to improve maintainability and performance.
> ** Server side cache implementation was refactored to use Scaffeine, a thin Scala wrapper for Caffeine which is very well known and stable Java caching library offering a great performance and support.
> ** Service log level was changed from DEBUG to INFO, to reduce logs verbosity and gain more clarity.
> 
> For information on:
> 
> * The **Enhanced Media Embed** plugin, see: xref:introduction-to-mediaembed.adoc[Enhanced Media Embed plugin].
> * The **Export** plugin, see: xref:export.adoc[Export plugin].
> * The **Enhanced Image Editing** plugin, see: xref:editimage.adoc[Enhanced Image Editing plugin].
> * The **Link Checker** plugin, see: xref:linkchecker.adoc[Link Checker plugin].
> * The **Spell Checker Pro** plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro plugin].
> * The **Spelling Autocorrect** plugin, see: xref:autocorrect.adoc[Spelling Autocorrect plugin].
> * Deploying the **server-side** components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
> 
> The Java server-side components have been updated to the following versions:
> 
> * `ephox-hyperlinking.war`: 2.105.20
> * `ephox-image-proxy.war`: 2.106.8
> * `ephox-spelling.war`: 2.118.15
> 
> === Updating the self-hosted server-side components

Author: ```suggestion
* **Enhanced Media Embed** plugin `mediaembed`.
* **Export** plugin `export`.
* **Enhanced Image Editing** plugin `editimage`.
* **Link Checker** plugin `linkchecker`.
* **Spell Checker Pro** plugin `tinymcespellchecker`.
* **Spelling Autocorrect** plugin `autocorrect`.

For information on:

* The **Enhanced Media Embed** plugin, see: xref:introduction-to-mediaembed.adoc[Enhanced Media Embed plugin].
* The **Export** plugin, see: xref:export.adoc[Export plugin].
* The **Enhanced Image Editing** plugin, see: xref:editimage.adoc[Enhanced Image Editing plugin].
* The **Link Checker** plugin, see: xref:linkchecker.adoc[Link Checker plugin].
* The **Spell Checker Pro** plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Spell Checker Pro plugin].
* The **Spelling Autocorrect** plugin, see: xref:autocorrect.adoc[Spelling Autocorrect plugin].
* Deploying the **server-side** components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].

The Java server-side components have been updated to the following versions:

* `ephox-hyperlinking.war`: 2.105.20
* `ephox-image-proxy.war`: 2.106.8
* `ephox-spelling.war`: 2.118.15

NOTE: Service log level was changed from DEBUG to INFO, to reduce logs verbosity and gain more clarity.

=== Updating the self-hosted server-side components
```

* xref:bug-fixes[Bug fixes]

[[accompanying-premium-skins-and-icon-packs-changes]]
== Accompanying Premium Skins and Icon Packs changes

The {productname} 6.7.2 release includes an accompanying release of the **Premium Skins and Icon Packs**.

=== Premium Skins and Icon Packs

The **Premium Skins and Icon Packs** release includes the following updates:

The **Premium Skins and Icon Packs** were rebuilt to pull in the changes also incorporated into the default {productname} 6.7.2 skin, Oxide.

For information on using premium skins and icon packs, see: xref:premium-skins-and-icons.adoc[Premium Skins and Icon Packs].

[[bug-fixes]]
== Bug fixes

{productname} 6.7.2 also includes the following bug fixes:

=== Applying heading formatting to the content of the `summary` element extended its application to the content of the parent `details` element.
// TINY-10154

// CCFR

=== The function `getModifierState` did not work on events passed through the editor as expected.
// TINY-10263

// CCFR

=== List items containing a nested list between two text caused some list operations to fail.
// TINY-10268

// CCFR

> === css loading in `toolbardrawertoggletest` flickering.

Author: ```suggestion
=== Removed use of async for editor rendering which caused visual blinking when reloading the editor in-place.
```

= TinyMCE 6.7.2
:navtitle: TinyMCE 6.7.2
:description: Release notes for TinyMCE 6.7.2
:keywords: releasenotes, new, changes, bugfixes
:page-toclevels: 1

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.7.2 was released for {enterpriseversion} and {cloudname} on Wednesday, November 1^st^, 2023. These release notes provide an overview of the changes for {productname} 6.7.2, including:

> 
> * xref:bug-fixes[Bug fixes]

Author: ```suggestion

* xref:accompanying-premium-self-hosted-server-side-component-changes[Accompanying Premium self-hosted server-side component changes]
* xref:bug-fixes[Bug fixes]
```

= TinyMCE 6.7.2
:navtitle: TinyMCE 6.7.2
:description: Release notes for TinyMCE 6.7.2
:keywords: releasenotes, new, changes, bugfixes
:page-toclevels: 1

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.7.2 was released for {enterpriseversion} and {cloudname} on Wednesday, November 1^st^, 2023. These release notes provide an overview of the changes for {productname} 6.7.2, including:

* xref:accompanying-premium-self-hosted-server-side-component-changes[Accompanying Premium self-hosted server-side component changes]
* xref:bug-fixes[Bug fixes]

> 
> [[bug-fixes]]

Author: ```suggestion

[[accompanying-premium-self-hosted-server-side-component-changes]]
== Accompanying Premium self-hosted server-side component changes

The {productname} 6.7.2 release includes accompanying changes affecting the {productname} **self-hosted** services for the following plugins:

* 
*
*
*
////
Example ONLY
The Java server-side components have been updated to the following versions:

* `ephox-hyperlinking.war`: 2.105.13
* `ephox-image-proxy.war`: 2.106.4
* `ephox-spelling.war`: 2.118.6
////
For information on:

////
Example ONLY 
* The Enhanced Media Embed plugin, see: xref:introduction-to-mediaembed.adoc[Enhanced Media Embed plugin].
*
*
*
////
=== Updating the self-hosted server-side components
////
The new versions of the server-side services provide updates for the Java-based server-side components. To deploy the updated version of the server-side components:

. Update your Java Application Server to the minimum required version:

include::partial$misc/supported-application-servers.adoc[]

. Replace the existing server-side `.war` files with the `.war` files bundled with {productname} 6.2 or later.

For information on:

* Deploying the server-side components, see: xref:introduction-to-premium-selfhosted-services.adoc[Server-side component installation].
* Deploying the server-side components using Docker, see: xref:bundle-intro-setup.adoc[Containerized service deployments].
////

[[bug-fixes]]
```

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.7.2 was released for {enterpriseversion} and {cloudname} on Wednesday, November 1^st^, 2023. These release notes provide an overview of the changes for {productname} 6.7.2, including:

* xref:bug-fixes[Bug fixes]

[[bug-fixes]]
== Bug fixes

{productname} 6.7.2 also includes the following bug fixes:

=== Applying heading formatting to the content of the `summary` element extended its application to the content of the parent `details` element.
// TINY-10154

In previous versions of {productname}, the accordion summary was not functioning as expected.

As a consequence, a structural inconsistency was identified in the editor schema, where heading elements were erroneously considered **invalid** children of a `<summary>` node and any attempts to apply heading formatting to content within the `<summary>` proved unsuccessful.

{productname} 6.7.2 addresses this issue, now, the editor schema permits heading elements as **valid** children of the summary element.

As a result, the `<summary>` element now accommodates phrasing content or a single heading element as expected.

> 

Author: @vpyshnenko and @spocke see below workaround suggestion:
```suggestion

IMPORTANT: After applying heading tags (<h1>-<h6>) to a context selection within a <summary> element, it's important to note that in order to revert the formatting to its original state, the context selection must be reselected within the <summary> element, then the user must reapply aka **un-check** the heading format from within either the menubar>format selection or the toolbar>formats selection to remove them. Currently, there are **no alternative** workarounds available for reversing this formatting within a <summary> element.

```

> === Applying heading formatting to the content of the `summary` element extended its application to the content of the parent `details` element.
> // TINY-10154
> 
> In previous versions of {productname}, the accordion summary was not functioning as expected.
> 
> As a consequence, a structural inconsistency was identified in the editor schema, where heading elements were erroneously considered **invalid** children of a `<summary>` node and any attempts to apply heading formatting to content within the `<summary>` proved unsuccessful.
> 
> {productname} 6.7.2 addresses this issue, now, the editor schema permits heading elements as **valid** children of the summary element.
> 
> As a result, the `<summary>` element now accommodates phrasing content or a single heading element as expected.
> 

Author: ```suggestion
```
